catalog/app/containers/Auth/SSOSignUp.js

```diff
@@ -0,0 +1,148 @@
+import { goBack, push } from 'connected-react-router'
+import * as React from 'react'
+import { FormattedMessage as FM } from 'react-intl'
+import { Redirect } from 'react-router-dom'
+import { reduxForm, Field, SubmissionError } from 'redux-form/es/immutable'
+import * as reduxHook from 'redux-react-hook'
+import * as M from '@material-ui/core'
+import { unstable_Box as Box } from '@material-ui/core/Box'
+
+import { push as notify } from 'containers/Notifications/actions'
+import * as Config from 'utils/Config'
+import * as NamedRoutes from 'utils/NamedRoutes'
+import * as Sentry from 'utils/Sentry'
+import Link from 'utils/StyledLink'
+import defer from 'utils/defer'
+import parseSearch from 'utils/parseSearch'
+import * as validators from 'utils/validators'
+
+import * as Layout from './Layout'
+import * as actions from './actions'
+import * as errors from './errors'
+import msg from './messages'
+import * as selectors from './selectors'
+
+const Container = Layout.mkLayout(<FM {...msg.ssoSignUpHeading} />)
+
+const Form = reduxForm({ form: 'Auth.SSO.SignUp' })(({ children, ...props }) => (
+  <form onSubmit={props.handleSubmit}>{children(props)}</form>
+))
+
+export default ({ location: { search } }) => {
+  const { provider, token, next } = parseSearch(search)
+
+  const dispatch = reduxHook.useDispatch()
+  const { urls } = NamedRoutes.use()
+  const sentry = Sentry.use()
+  const authenticated = reduxHook.useMappedState(selectors.authenticated)
+  const cfg = Config.useConfig()
+
+  const back = React.useCallback(() => dispatch(goBack()), [dispatch])
+
+  const onSubmit = React.useCallback(
+    async (values) => {
+      const { username } = values.toJS()
+      try {
+        const result = defer()
+        dispatch(actions.signUp({ username, provider, token }, result.resolver))
+        await result.promise
+      } catch (e) {
+        if (e instanceof errors.UsernameTaken) {
+          throw new SubmissionError({ username: 'taken' })
+        }
+        if (e instanceof errors.InvalidUsername) {
+          throw new SubmissionError({ username: 'invalid' })
+        }
+        if (e instanceof errors.EmailDomainNotAllowed) {
+          throw new SubmissionError({ _error: 'emailDomain' })
```

Comment:
We display the error message in the form that's asking the user to enter a username - but, it's too late at that point; no username will make the error go away. @akarve: I'm guessing it's not a big deal, but just wanted to point it out.

Reply:
@dimaryaz you can add this check to the sign-in request handler, so i can catch the error before redirecting to this form

Reply:
I could do that... but, right now, if the user exists, we don't do any more checks. (The admin could've added the user manually, regardless of domain restrictions, or whether signups are allowed at all.)

For now, let's not worry about this.

Reply:
when the user exists, it's not required. it's required when user doesn't exist, like, you can check if the email has valid domain and add some extra message / prop when throwing `'User does not exist'` api exception